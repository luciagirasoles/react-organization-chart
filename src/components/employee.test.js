import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import Employee from "./employee";

test("Employee", async () => {
  const onNewSubordinate = jest.fn();

  const id = new Date("2019-06-10T03:25:39.064Z").getTime();
  const employeeID = id + 1;
  const employees = {
    [id]: {
      id,
      name: "John Doe",
      title: "CEO",
      subordinates: [employeeID]
    },
    [employeeID]: {
      id: employeeID,
      name: "Jane Doe",
      title: "CTO",
      subordinates: []
    }
  };

  const { getByLabelText, asFragment } = render(
    <Employee
      {...employees[id]}
      level={0}
      employees={employees}
      companyName="ACME Inc."
      onNewSubordinate={onNewSubordinate}
      highlightedTitle="CTO"
    />
  );

  expect(asFragment()).toMatchSnapshot();

  const button = getByLabelText("Add new employee under John Doe");

  fireEvent.click(button);

  expect(asFragment()).toMatchSnapshot();

  const employeeInput = await waitForElement(() =>
    getByLabelText("Write the employee name")
  );
  fireEvent.change(employeeInput, { target: { value: "John Wick" } });

  const titleInput = getByLabelText("Write the employee title");
  fireEvent.change(titleInput, { target: { value: "COO" } });

  fireEvent.submit(titleInput);

  expect(onNewSubordinate).toHaveBeenCalledWith({
    id,
    title: "COO",
    name: "John Wick"
  });

  fireEvent.click(button);

  fireEvent.keyDown(document, {
    key: "Escape",
    code: 27,
    charCode: 27
  });

  expect(asFragment()).toMatchSnapshot();
});
