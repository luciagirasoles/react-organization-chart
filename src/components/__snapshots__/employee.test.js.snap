// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee 1`] = `
<DocumentFragment>
  .emotion-3 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: white;
  color: black;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
}

.emotion-2 {
  font-size: .8rem;
  font-weight: lighter;
  margin: 0;
}

.emotion-8 {
  margin: 1rem 0 0 3rem;
}

.emotion-7 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: #028265;
  color: white;
}

.emotion-4 {
  background: none;
  border: none;
  color: white;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

<div>
    <div
      class="emotion-3"
      id="1560137139064"
    >
      <button
        aria-label="Add new employee under John Doe"
        class="emotion-0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
      <h2
        class="emotion-1"
      >
        John Doe
      </h2>
      <h3
        class="emotion-2"
      >
        Title: 
        <strong>
          CEO
        </strong>
      </h3>
    </div>
    <section
      class="emotion-8"
    >
      <div>
        <div
          class="emotion-7"
          id="1560137139065"
        >
          <button
            aria-label="Add new employee under Jane Doe"
            class="emotion-4"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </button>
          <h2
            class="emotion-1"
          >
            Jane Doe
          </h2>
          <h3
            class="emotion-2"
          >
            Title: 
            <strong>
              CTO
            </strong>
          </h3>
        </div>
        <section
          class="emotion-8"
        />
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Employee 2`] = `
<DocumentFragment>
  .emotion-3 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: white;
  color: black;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
}

.emotion-2 {
  font-size: .8rem;
  font-weight: lighter;
  margin: 0;
}

.emotion-8 {
  margin: 1rem 0 0 3rem;
}

.emotion-7 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: #028265;
  color: white;
}

.emotion-4 {
  background: none;
  border: none;
  color: white;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

<div>
    <div
      class="emotion-3"
      id="1560137139064"
    >
      <button
        aria-label="Add new employee under John Doe"
        class="emotion-0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
      <h2
        class="emotion-1"
      >
        John Doe
      </h2>
      <h3
        class="emotion-2"
      >
        Title: 
        <strong>
          CEO
        </strong>
      </h3>
    </div>
    <section
      class="emotion-8"
    >
      <div>
        <div
          class="emotion-7"
          id="1560137139065"
        >
          <button
            aria-label="Add new employee under Jane Doe"
            class="emotion-4"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </button>
          <h2
            class="emotion-1"
          >
            Jane Doe
          </h2>
          <h3
            class="emotion-2"
          >
            Title: 
            <strong>
              CTO
            </strong>
          </h3>
        </div>
        <section
          class="emotion-8"
        />
      </div>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Employee 3`] = `
<DocumentFragment>
  .emotion-3 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: white;
  color: black;
}

.emotion-0 {
  background: none;
  border: none;
  color: black;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

.emotion-1 {
  font-size: 1rem;
  font-weight: bold;
  margin: 0;
  margin-bottom: .5rem;
}

.emotion-2 {
  font-size: .8rem;
  font-weight: lighter;
  margin: 0;
}

.emotion-8 {
  margin: 1rem 0 0 3rem;
}

.emotion-7 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  display: inline-block;
  padding: 1rem 3rem;
  width: auto;
  position: relative;
  background-color: #028265;
  color: white;
}

.emotion-4 {
  background: none;
  border: none;
  color: white;
  cursor: pointer;
  position: absolute;
  top: .5rem;
  right: .5rem;
  outline: none;
}

<div>
    <div
      class="emotion-3"
      id="1560137139064"
    >
      <button
        aria-label="Add new employee under John Doe"
        class="emotion-0"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          />
        </svg>
      </button>
      <h2
        class="emotion-1"
      >
        John Doe
      </h2>
      <h3
        class="emotion-2"
      >
        Title: 
        <strong>
          CEO
        </strong>
      </h3>
    </div>
    <section
      class="emotion-8"
    >
      <div>
        <div
          class="emotion-7"
          id="1560137139065"
        >
          <button
            aria-label="Add new employee under Jane Doe"
            class="emotion-4"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </button>
          <h2
            class="emotion-1"
          >
            Jane Doe
          </h2>
          <h3
            class="emotion-2"
          >
            Title: 
            <strong>
              CTO
            </strong>
          </h3>
        </div>
        <section
          class="emotion-8"
        />
      </div>
    </section>
  </div>
</DocumentFragment>
`;
